src/NServiceBus.Transport.SQS/HybridPubSubChecker.cs

```diff
@@ -71,34 +48,53 @@ public async Task<bool> ThisIsAPublishMessageNotUsingMessageDrivenPubSub(Unicast
                 && unicastTransportOperation.Message.GetMessageIntent() == MessageIntent.Publish)
             {
                 var eventType = multicastEventsMessageIdsToType[unicastTransportOperation.Message.MessageId];
-                var existingTopic = await topicCache.GetTopic(eventType).ConfigureAwait(false);
+                var existingTopic = await topicCache.GetTopic(eventType, cancellationToken).ConfigureAwait(false);
                 if (existingTopic == null)
                 {
                     return false;
                 }
 
                 var cacheKey = existingTopic.TopicArn + unicastTransportOperation.Destination;
-                Logger.Debug($"Performing first subscription cache lookup for '{cacheKey}'.");
-                if (!TryGetFromCache(cacheKey, out var cacheItem))
+                if (Logger.IsDebugEnabled)
                 {
-                    cacheItem = await rateLimiter.Execute(async () =>
+                    Logger.Debug($"Performing subscription cache lookup for '{cacheKey}'.");
+                }
+
+                var lazyCacheItem = subscriptionsCache.AddOrUpdate(cacheKey,
```

Comment:
Do I understand it correctly, that `AddOrUpdate` will synchronize the creation of the `LazyCacheItem`s. Eg. if there are two queries for the same topic, the first to execute `AddOrUpdate` will schedule `LazyCacheItem` creation, and the second one will get to the `Update` lambda, check that the entry is fresh, and return what the first has grabbed from SNS.